### ISSUE TITLE
Training Process Issues

### ISSUE THREAD

**Issue by xiaoshuai0719** [sunshangquan/Histoformer#23]
Hello author, Thank you for your work.I encountered some issues during training. In the yml dataset section, I only kept the training dataset, and I commented out the sections for the validation datasets. May I ask whether in the yaml section I should keep one validation set or maintain the code settings for all validation set parts during the training process?Here are the parts that I commented out during training:
#  val_snow_s:
#    name: ValSet_Snow100K-S
#    type: Dataset_PairedImage
#    dataroot_gt: "E:/Snow100K-Test/Snow100K-S/gt"
#    dataroot_lq: "E:/Snow100K-Test/Snow100K-S/synthetic"
#    io_backend:
#      type: disk
#  val_snow_l:
#    name: ValSet_Snow100K-L
#    type: Dataset_PairedImage
#    dataroot_gt: /home1/ssq/data/allweather/test/Snow100K-L/gt/
#    dataroot_lq: /home1/ssq/data/allweather/test/Snow100K-L/synthetic/
#    io_backend:
#      type: disk
#  val_test1:
#    name: ValSet_Test1
#    type: Dataset_PairedImage
#    dataroot_gt: /home1/ssq/data/allweather/test/Test1/gt/
#    dataroot_lq: /home1/ssq/data/allweather/test/Test1/input/
#    io_backend:
#      type: disk
#  val_raindrop:
#    name: ValSet_RainDrop
#    type: Dataset_PairedImage
#    dataroot_gt: /home1/ssq/data/allweather/test/RainDrop/gt/
#    dataroot_lq: /home1/ssq/data/allweather/test/RainDrop/input/
#    io_backend:
#      type: disk

**Comment by sunshangquan** [comment]
Hi @xiaoshuai0719 , thank you for your attention. The validation sets can be changed by also changing Lines 115-162 in the [train.py](https://github.com/sunshangquan/Histoformer/blob/1f045f06c03551c31504d8042dbe6ff9b9569108/basicsr/train.py#L115). If you want to remove the entire validation process, you could set whatever your validation set is (for example a single identical image for both input and ground-truth) and change the value of ```60``` [here](https://github.com/sunshangquan/Histoformer/blob/1f045f06c03551c31504d8042dbe6ff9b9569108/basicsr/models/image_restoration_model.py#L272) to ```0``` to skip validation process. If this cannot address your issue, could you further show the specific error information you encountered?

**Comment by xiaoshuai0719** [comment]
Hello author, I would like to know if I need to comment out the validation dataset part during the process of running train.py, or should I only leave the single validation set that I need, in other words, do I need to use the validation set val during the training process?

**Comment by xiaoshuai0719** [comment]
In other words, do I need to comment out the val dataset settings in the Allweather_Histoformer.yml during the training process?

**Comment by xiaoshuai0719** [comment]
When I first started training, I commented out the 'val' section in the Allweather_Histoformer.yml file (as shown in the figure below), so for normal training, I should also ensure that the val dataset is loaded properly without commenting out, so that I can get the best.pth, right?

![屏幕截图 2024-12-17 112323](https://github.com/user-attachments/assets/bc680c34-baed-4b9a-8e46-1a580bfc244f)

**Comment by sunshangquan** [comment]
Indeed, you may safely try it and comment it out for your intended purpose.
